onnxruntime/core/mlas/lib/qladd.cpp

```diff
@@ -533,17 +532,93 @@ MlasQLinearAddKernelHelper(
         // Pack, saturate, and add output zero point.
         const int16x8_t vacc0 = vqaddq_s16(vcombine_s16(vqmovn_s32(vacc0_lo), vqmovn_s32(vacc0_hi)), VectorZeroPointC);
         const int16x8_t vacc1 = vqaddq_s16(vcombine_s16(vqmovn_s32(vacc1_lo), vqmovn_s32(vacc1_hi)), VectorZeroPointC);
-        vc = SUI::combine_i8_s16(vacc0, vacc1);
+        typename SUI::i8x16_t vc = SUI::combine_i8_s16(vacc0, vacc1);
 
         n -= 16;
-        if (n < 0) break;
-
         SUI::vst1q_i8(OutputC, vc);
         OutputC += 16;
     }
 
-    if (n < 0) {
-        n += 16;
+    if (n > 0) {
+        typename SUI::T TailDataA[16] = { 0 };
+        typename SUI::T TailDataB[16] = { 0 };
+
+        if (!IsScalarA) {
+            typename SUI::T* tail = TailDataA;
```

Comment:
Can this be an inline function to avoid code duplication?

Reply:
done!